src/main/java/seedu/address/logic/filters/CoeExpiryFilter.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.logic.filters;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import seedu.address.model.customer.CoeExpiry;
+import seedu.address.model.customer.Customer;
+
+public class CoeExpiryFilter extends AbstractFilter {
+    public static final String MESSAGE_CONSTRAINTS = "COE Expiry Filter can only accept 'exp' or a non negative number";
+    private final LocalDate coeExpiryThreshold;
+
+    /**
+     * Create a filter for COE Expiry, checking whether a COE
+     * expires in {@code filterString} years or less.
+     *
+     * A special keyword "exp" is accepted as an alias for COE
+     * that expires in zero years or less (ie: Already expired).
+     *
+     * Note that {@code filterString} must either be "exp" or a non negative number.
+     *
+     * @param filterString The number of years that a COE can be valid for before
+     *                     being skipped.
+     */
+    public CoeExpiryFilter(String filterString) {
```

Comment:
good idea would be to trim the filterString that you get

Reply:
Will put the trimming function in the ParseUtil class.